powertools-idempotency/src/main/java/software/amazon/lambda/powertools/idempotency/IdempotencyConfig.java

```diff
@@ -0,0 +1,215 @@
+         */
+        public IdempotencyConfig build() {
+            return new IdempotencyConfig(
+                    eventKeyJMESPath,
+                    payloadValidationJMESPath,
+                    throwOnNoIdempotencyKey,
+                    useLocalCache,
+                    localCacheMaxItems,
+                    expirationInSeconds,
+                    hashFunction);
+        }
+
+        /**
+         * A JMESPath expression to extract the idempotency key from the event record. <br>
+         * See <a href="https://jmespath.org/">https://jmespath.org/</a> for more details.<br>
+         * Common paths are: <ul>
+         *     <li><code>powertools_json(body)</code> for APIGatewayProxyRequestEvent and APIGatewayV2HTTPEvent</li>
+         *     <li><code>Records[*].powertools_json(body)</code> for SQSEvent</li>
+         *     <li><code>Records[0].Sns.Message | powertools_json(@)</code> for SNSEvent</li>
+         *     <li><code>detail</code> for ScheduledEvent (EventBridge / CloudWatch events)</li>
+         *     <li><code>Records[*].kinesis.powertools_json(powertools_base64(data))</code> for KinesisEvent</li>
+         *     <li><code>Records[*].powertools_json(powertools_base64(data))</code> for KinesisFirehoseEvent</li>
+         *     <li>...</li>
+         * </ul>
+         *
+         *
+         * @param eventKeyJMESPath path of the key in the Lambda event
+         * @return the instance of the builder (to chain operations)
+         */
+        public Builder withEventKeyJMESPath(String eventKeyJMESPath) {
+            this.eventKeyJMESPath = eventKeyJMESPath;
+            return this;
+        }
+
+        /**
+         * Set the maximum number of items to store in local cache, by default 256
+         *
+         * @param localCacheMaxItems maximum number of items to store in local cache
+         * @return the instance of the builder (to chain operations)
+         */
+        public Builder withLocalCacheMaxItems(int localCacheMaxItems) {
+            this.localCacheMaxItems = localCacheMaxItems;
+            return this;
+        }
+
+        /**
+         * Whether to locally cache idempotency results, by default true
+         *
+         * @param useLocalCache boolean that indicate if a local cache must be used in addition to the persistence store.
+         *                      If set to true, will use the {@link LRUCache}
+         * @return the instance of the builder (to chain operations)
+         */
+        public Builder withUseLocalCache(boolean useLocalCache) {
+            this.useLocalCache = useLocalCache;
+            return this;
+        }
+
+        /**
+         * The number of seconds to wait before a record is expired
+         *
+         * @param expirationInSeconds expiration of the record in the store (in seconds)
+         * @return the instance of the builder (to chain operations)
+         */
+        public Builder withExpirationInSeconds(int expirationInSeconds) {
```

Comment:
Would it makes sense to have this api has `withExpiraton` and  have java.time.Duration as parameter ?

Reply:
Why not. Will be simpler to read.